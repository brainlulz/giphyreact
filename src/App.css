@font-face {
  font-family: 'Gears of Peace';
  src: url('./public/GearsofPeace.woff2') format('woff2'),
      url('./public/GearsofPeace.woff') format('woff');
  font-weight: 500;
  font-style: normal;
}

@font-face {
  font-family: 'SubZER0';
  src: url('./public/SubZER0.woff2') format('woff2'),
      url('./public/SubZER0.woff') format('woff');
  font-weight: normal;
  font-style: normal;
}

.App {
  font-family: 'SubZER0';  
  display: grid;
  grid-template-columns: 5rem 1fr 5rem;
  grid-auto-rows: minmax(5rem, auto);
}
