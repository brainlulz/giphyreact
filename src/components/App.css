@font-face {
  font-family: 'Gears of Peace';
  src: url('../assets/fonts/GearsofPeace.woff2') format('woff2'),
      url('../assets/fonts/GearsofPeace.woff') format('woff');
  font-weight: 500;
  font-style: normal;
}

@font-face {
  font-family: 'SubZER0';
  src: url('../assets/fonts/SubZER0.woff2') format('woff2'),
      url('../assets/fonts/SubZER0.woff') format('woff');
  font-weight: normal;
  font-style: normal;
}

.App {
  font-family: 'SubZER0';  
  display: grid;
  grid-template-columns: 5rem 1fr 5rem;
  grid-auto-rows: minmax(5rem, auto);
}
