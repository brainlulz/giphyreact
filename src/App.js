import React, { Component } from "react";
import { Route, withRouter, Switch } from "react-router-dom";
import "./App.css";

import Search from "./components/Search";
//import Results from "./components/Results";
import ResultsNumber from "./components/Results/ResultsNumber";
import Error from "./components/Error";
import NoMatch from "./components/UI/NoMatch/NoMatch";
import Home from "./components/UI/Home/Home";
//import favoritesUtils from "./Utilities/favoritesUtils";
import Header from "./components/Header";
import ResultsGifs from "./container/ResultsGifs";

const giphUrl = `http://api.giphy.com/v1/gifs/search?q`;
const apiKey = `vw4usscjAkDQiPbUvnGdRJfpEUWqZsuY`;

class App extends Component {
  constructor() {
    super();
    this.state = {
      value: "",
      gifs: [],
      error: false,
      loading: false
      /* favorites: [] */
    };
  }
  componentDidMount(event) {
    /* this.setState({
      favorites: favoritesUtils.loadFavorites()
    }); */
    /* if (this.props.history.location.pathname === "/favorited") {
      this.handleFavorited();
    } */
    if (this.props.location.search) {
      this.handleSearchSubmit(this.props.location.search);
    }
  }

  componentWillReceiveProps(nextProps) {
    if (
      nextProps.location.search &&
      nextProps.location.search !== this.props.location.search
    ) {
      this.handleSearchSubmit(nextProps.location.search);
    }
  }

  handleSearchSubmit(search) {
    const queryParams = new URLSearchParams(search);
    const searchedText = queryParams.get("q");
    this.setState({ value: searchedText });
    this.handleSubmit(null, searchedText);
  }

  handleSubmit = (event, value) => {
    if (event) {
      event.preventDefault();
    }
    fetch(`${giphUrl}=${value}&api_key=${apiKey}`)
      .then(response => {
        if (response.status === 200) {
          return response.json();
        }
        return Promise.reject({ status: response.status });
      })
      .then(data => {
        const recievedData = data.data;
        this.setState({ gifs: recievedData });
      })
      .catch(error => {
        this.setState({ error: true });
        this.props.history.push("/error");
      });
  };

  handleChange = event => {
    this.setState({ value: event.target.value });
  };

  /* handleFavorited = () => {
    const gifStored = JSON.parse(localStorage.getItem("gifs"));
    if (gifStored !== null) {
      this.setState({ gifs: gifStored });
    }
  }; */

  /* handleToggleFavorite = favorite => {
    const { favorites } = this.state;
    const isFavorite = favorites.find(fav => fav.id === favorite.id);
    let nextFavorites;
    if (isFavorite) {
      // remove favorite
      nextFavorites = favorites.filter(fav => fav.id !== favorite.id);
    } else {
      // add favorite
      nextFavorites = [...favorites, favorite];
    }
    favoritesUtils.saveFavorites(nextFavorites);
    this.setState({ favorites: nextFavorites });
  }; */

  render() {
    //const { favorites } = this.state;
    return (
      <div className="App">
        <Header favClicked={this.handleFavorited} />
        <Search
          value={this.state.value}
          onSubmit={this.handleSubmit}
          onChange={this.handleChange}
        />
        <Switch>
          <Route exact path="/" component={Home} />
          <Route
            exact
            path="/favorited"
            render={() => (
              <ResultsGifs
                resultType="favorites"
                onToggleFavorite={this.handleToggleFavorite}
              />
            )}
          />
          <Route
            exact
            path={`/search/`}
            render={() => (
              <React.Fragment>
                <ResultsNumber number={this.state.gifs.length} />
                <ResultsGifs
                  resultType="search"
                  searchResults={this.state.gifs}
                  onToggleFavorite={this.handleToggleFavorite}
                />
              </React.Fragment>
            )}
          />
          <Route exact path={"/error"} component={Error} />
          <Route component={NoMatch} />
        </Switch>
      </div>
    );
  }
}

export default withRouter(App);
