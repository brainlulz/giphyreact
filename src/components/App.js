import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import './App.css';

import Error from './Error';
import NoMatch from './UI/NoMatch/NoMatch';
import Home from './UI/Home/Home';
import Header from './Header';
import ResultsGifs from '../container/ResultsGifs';
import SearchGifs from '../container/SearchGifs';

const getValueFromQuerySearchParams = search => {
  const queryParams = new URLSearchParams(search);
  return queryParams.get('q');
};

class App extends Component {
  componentDidMount(event) {
    if (this.props.location.search) {
      const value = getValueFromQuerySearchParams(this.props.location.search);
      this.props.changeValue(value);
      this.props.getGifs(value);
    }
  }

  componentWillReceiveProps(nextProps) {
    if (
      nextProps.location.search &&
      nextProps.location.search !== this.props.location.search
    ) {
      const value = getValueFromQuerySearchParams(nextProps.location.search);
      this.props.changeValue(value);
      this.props.getGifs(value);
    }
  }

  render() {
    return (
      <div className="App">
        <Header />
        <SearchGifs />
        <Switch>
          <Route exact path="/" component={Home} />
          <Route
            exact
            path="/favorited"
            render={() => (
              <ResultsGifs
                resultType="favorites"
                onToggleFavorite={this.handleToggleFavorite}
              />
            )}
          />
          <Route
            exact
            path={`/search`}
            render={() => (
              <React.Fragment>
                {/* <ResultsNumber number={this.props.search.results.length} /> */}
                <ResultsGifs
                  resultType="search"
                  searchResults={this.results}
                  onToggleFavorite={this.handleToggleFavorite}
                />
              </React.Fragment>
            )}
          />
          <Route exact path={'/error'} component={Error} />
          <Route component={NoMatch} />
        </Switch>
      </div>
    );
  }
}

export default App;
