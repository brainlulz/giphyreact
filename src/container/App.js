import { connect } from 'react-redux';
import App from '../components/App';
import { withRouter } from 'react-router-dom';
import { getGifs, changeValue } from '../actions';

const mapStateToProps = state => {
  return {
      value: state.search.value
  };
};

const mapDispatchToProps = { getGifs, changeValue };

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));