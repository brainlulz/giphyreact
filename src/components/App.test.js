import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  it('renders the App component', () => {
    const wrapper = shallow(<App location={{search: undefined}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
