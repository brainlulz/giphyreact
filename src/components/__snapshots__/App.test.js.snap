// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the App component 1`] = `
<div
  className="App"
>
  <header />
  <withRouter(Connect(Search)) />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      exact={true}
      path="/favorited"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/search"
      render={[Function]}
    />
    <Route
      component={[Function]}
      exact={true}
      path="/error"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
